stats/stats/src/data_source/kinds/local_db/mod.rs

```diff
@@ -150,7 +150,9 @@ where
     ) -> Result<(), UpdateError> {
         let metadata = get_chart_metadata(cx.db, ChartProps::NAME).await?;
         if let Some(last_updated_at) = metadata.last_updated_at {
-            if cx.time == last_updated_at {
+            // DB stores timestamps with microsecond precision
+            let update_time = cx.time.trunc_subsecs(6);
+            if update_time == last_updated_at {
```

Comment:
why not just `if update_time >= last_update_at` ? you wont need to truncate anything

Reply:
`>=` is just incorrect, because we don't want to skip all updates in the future
`<=` won't work because after truncating (=`last_updated_at`), we have `cx.time >= last_updated_at`

Reply:
okay, when let's truncate until seconds and compare only seconds value. if time difference is less than a second, I don't think we might want to perform update

Reply:
decided to extract into a separate function for better readability